### PR TITLE
feat: add polymophism support

### PR DESCRIPTION
**Pull request by jbarreau**

### CODE REVIEW

File: src/datasource_toolkit/forestadmin/datasource_toolkit/decorators/search/collections.py [hunk 1]
```diff
@@ -124,10 +127,19 @@ def _get_searchable_fields(self, collection: Collection, extended: bool) -> List
...
+            if extended and (is_many_to_one(field) or is_one_to_one(field) or is_polymorphic_one_to_one(field)):
                 related = collection.datasource.get_collection(field["foreign_collection"])
 
                 for sub_name, sub_field in related.schema["fields"].items():
                     if is_column(sub_field):
                         fields.append((f"{name}:{sub_name}", sub_field))
+
+            if extended and is_polymorphic_many_to_one(field):
+                ForestLogger.log(
+                    "info",
```
Comment (jbarreau) [inline src/datasource_toolkit/forestadmin/datasource_toolkit/decorators/search/collections.py:139]: should it be a debug log ?

File: src/datasource_toolkit/forestadmin/datasource_toolkit/decorators/computed/collections.py [hunk 2]
```diff
@@ -24,15 +31,18 @@ def __init__(self, *args: Any, **kwargs: Any):
...
+    def get_computed(self, path_p: str) -> Union[ComputedDefinition, None]:
+        if ":" not in path_p:
+            return self._computeds.get(path_p)
 
-        related_field, path = path.split(":", 1)
+        related_field, path = path_p.split(":", 1)
         field = cast(RelationAlias, self.get_field(related_field))
+        if is_polymorphic_many_to_one(field):
+            ForestLogger.log(
+                "info", f"Cannot compute computed fields over polymorphic relation {self.name}.{related_field}."
```
Comment (jbarreau) [inline src/datasource_toolkit/forestadmin/datasource_toolkit/decorators/computed/collections.py:42]: should be a debug log ?

File: src/datasource_toolkit/forestadmin/datasource_toolkit/decorators/search/collections.py (hunk 1 above)
Comment (nicolasalexandre9) [reply src/datasource_toolkit/forestadmin/datasource_toolkit/decorators/search/collections.py:139]: IMO, yes 👍

File: src/datasource_toolkit/forestadmin/datasource_toolkit/decorators/search/collections.py [hunk 3]
```diff
@@ -124,10 +127,19 @@ def _get_searchable_fields(self, collection: Collection, extended: bool) -> List
...
+                    f"We're not searching through {self.name}.{name} because it's a polymorphic relation. "
+                    "You can override the default search behavior with 'replace_search'."
+                    " See more: https://docs.forestadmin.com/developer-guide-agents-python/agent-customization/search",
```
Comment (nicolasalexandre9) [inline src/datasource_toolkit/forestadmin/datasource_toolkit/decorators/search/collections.py:142]: useless comma after "python/agent-customization/search" right ?

Comment (jbarreau) [reply src/datasource_toolkit/forestadmin/datasource_toolkit/decorators/search/collections.py:142]: yes you are right, but notice it's not in the string, it's a parameter separator. 
So do your job and let my code formatter do his.

File: src/agent_toolkit/forestadmin/agent_toolkit/utils/forest_schema/generator_field.py [hunk 4]
```diff
@@ -127,6 +135,9 @@ def build_one_to_one_schema(
             "validations": [],
             "reference": f"{foreign_collection.name}.{relation['origin_key']}",
         }
+        if is_polymorphic_one_to_one(relation):
```
Comment (jbarreau) [inline src/agent_toolkit/forestadmin/agent_toolkit/utils/forest_schema/generator_field.py:138]: Does this exists in ruby ??

File: src/datasource_django/forestadmin/datasource_django/utils/polymorphic_util.py [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+    is_reverse_polymorphic_relation,
+)
+from forestadmin.datasource_toolkit.interfaces.query.condition_tree.nodes.base import ConditionTree
+from forestadmin.datasource_toolkit.interfaces.query.condition_tree.nodes.branch import ConditionTreeBranch
+from forestadmin.datasource_toolkit.interfaces.query.condition_tree.nodes.leaf import ConditionTreeLeaf
+from forestadmin.datasource_toolkit.interfaces.query.projections import Projection
+from forestadmin.datasource_toolkit.interfaces.records import RecordsDataAlias
+
+
+class DjangoPolymorphismUtil:
```
Comment (jbarreau) [inline src/datasource_django/forestadmin/datasource_django/utils/polymorphic_util.py:16]: May be review this part, it's not easily understandable

File: src/datasource_toolkit/forestadmin/datasource_toolkit/decorators/computed/utils/flattener.py [hunk 6]
```diff
@@ -65,7 +65,7 @@ def unflatten(flats: FlatRecordList, projection: List[str]) -> List[RecordsDataA
     for record_index in range(num_records):
         for path_index, path in enumerate(projection):
             # When a marker is found, the parent is null.
-            parts = [*filter(lambda part: part != _MARKER_NAME, path.split(":"))]
+            parts = [*filter(lambda part: part not in (_MARKER_NAME, "*"), path.split(":"))]
```
Comment (jbarreau) [inline src/datasource_toolkit/forestadmin/datasource_toolkit/decorators/computed/utils/flattener.py:68]: There is no modification in flattener in ruby. Is this useless ??

Comment (nicolasalexandre9) [reply src/datasource_toolkit/forestadmin/datasource_toolkit/decorators/computed/utils/flattener.py:68]: Yes 👍

**Review by nicolasalexandre9 — APPROVED** [review]
👍

**Review by nicolasalexandre9 — APPROVED** [review]

**Pull request merged**